miniapp/include/dlaf/miniapp/options.h

```diff
@@ -0,0 +1,270 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2021, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+#include <algorithm>
+#include <cctype>
+#include <exception>
+#include <iostream>
+#include <string>
+#include <type_traits>
+
+#include <blas.hh>
+
+#include <hpx/program_options.hpp>
+
+#include <dlaf/common/assert.h>
+#include <dlaf/common/format_short.h>
+#include <dlaf/types.h>
+
+namespace dlaf::miniapp {
+inline Backend parseBackend(const std::string& backend) {
+  if (backend == "default")
+    return Backend::Default;
+  else if (backend == "mc")
+    return Backend::MC;
+  else if (backend == "gpu") {
+#if !defined(DLAF_WITH_CUDA)
+    std::cout << "Asked for --backend=gpu but DLAF_WITH_CUDA is disabled!" << std::endl;
+    std::terminate();
+#endif
+    return Backend::GPU;
+  }
+
+  std::cout << "Invalid option for --backend. Got '" << backend
+            << "' but expected one of 'default', 'mc', 'gpu'." << std::endl;
+  std::terminate();
+  return Backend::Default;  // unreachable
+}
+
+enum class SupportedTypes { OnlyReal, RealAndComplex };
+enum class ElementType { Single, Double, ComplexSingle, ComplexDouble };
+
+template <SupportedTypes types>
```

Comment:
Ok with me if you think it might be needed at some point. If yes, I'd do something like

```
enum class SupportReal { No, Yes };
enum class SupportComplex { No, Yes };
template <SupportReal, SupportComplex> struct MiniappOptions; 
```
so that usage of `MiniappOptions` is `MiniappOptions<SupportReal::Yes, SupportComplex::No>` instead of `MiniappOptions<true, false>`. Does that look ok to you including names?

Reply:
Not 100% sure that the only complex is needed, but there is a good chance.
Your suggestion of including names is even better.